vendor/src/github.com/docker/go-connections/nat/nat.go

```diff
@@ -146,9 +146,31 @@ func ParsePortSpecs(ports []string) (map[Port]struct{}, map[Port][]PortBinding,
 			rawPort = fmt.Sprintf(":%s", rawPort)
 		}
 
-		parts, err := PartParser(portSpecTemplate, rawPort)
-		if err != nil {
-			return nil, nil, err
+		parts := make(map[string]string)
+		if rawPort != "" && rawPort[0] == '[' {
```

Comment:
> I am trying to see if we do really need the square brackets.

Even if we don’t need them, I think it makes sense to use them. Using square brackets for separating IPv6 addresses from ports is what all other programs do, so there’s a strong argument for consistency.

> So wondering if we can make use of the standard net functions here, e.g. net. SplitHostPort()

I don’t see what using `net.SplitHostPort()` would buy us in this case. I don’t think it makes the code any shorter or handles any more cases than what we handle currently. If you disagree, can you provide a code snippet to illustrate what you mean please?